### DISCUSSION TITLE
How can I make issues path more readable?

### DISCUSSION THREAD

**Discussion by mapringg** [open-circle/valibot#248] [Q&A] [answered]
```javascript
issues: [
    {
      reason: 'type',
      validation: 'number',
      origin: 'value',
      message: 'Invalid type',
      input: 'Mega',
      issues: undefined,
      abortEarly: undefined,
      abortPipeEarly: undefined,
      skipPipe: undefined,
      path: [Array]
    },
    {
      reason: 'type',
      validation: 'number',
      origin: 'value',
      message: 'Invalid type',
      input: 'Mega',
      issues: undefined,
      abortEarly: undefined,
      abortPipeEarly: undefined,
      skipPipe: undefined,
      path: [Array]
    }
]
```

Is it possible to adjust the default output, so I do not have to format and have duplicate logs?
(I am currently using next.js and the error page automatically log the thrown error.)

**Comment by fabian-hiller** [comment, marked as answer]
Can you share your code with me? Seems to be a bug. There should not be duplicate issues. And you can format the `path` with `flatten` or with your own logic.

**Reply by fabian-hiller** [reply]
https://valibot.dev/guides/errors/#formatting

**Reply by mapringg** [reply]
I was stimulating a failed state of an array of values, so there were multiple issues.

```javascript
const dataSchema = object({
	_id: string(),
	id: string(),
	src: string(),
	tier: string(),
	name: string(),
	verified: boolean(),
	followers: number(),
	followerGrowth: number(),
	subscribers: optional(number()),
	subscribersGrowth: optional(number()),
	averageEngagement: number(),
	positiveSentiment: optional(number()),
	postActive: string(),
	base64: string(),
});
export const discoverResultsSchema = object({
	total: number(),
	data: array(dataSchema),
});
```

I am using `discoverResultsSchema` to parse my data from an api.

**Reply by fabian-hiller** [reply]
Ok. Then it's probably not a bug. Did my answer help you anyway?

**Reply by mapringg** [reply]
Yes, `flatten` makes formatting much easier, but an override of the default would have been nice.

Thank you for making such an easy to use library.

**Reply by fabian-hiller** [reply]
You can also look at [`flatten`](https://github.com/fabian-hiller/valibot/blob/main/library/src/error/flatten/flatten.ts) and build your own error formatter. As I wrote [here](https://github.com/fabian-hiller/valibot/issues/36#issuecomment-1722404925) I plan to improve the error message individualization before we reach v1.
